sopel/modules/currency.py

```diff
@@ -1,102 +1,189 @@
+    if 'date' in rates_fiat_json:
+        if time.time() - rates_fiat_json['date'] < 24 * 60 * 60:
+            return
+
+    # Update crypto rates
+    request = requests.get(crypto_url)
+    request.raise_for_status()
+    rates_btc_json = request.json()
+
+    # Update fiat rates
+    if bot.config.currency.fixer_io_key is not None:
+        request = requests.get(fixer_url.format(bot.config.currency.fixer_io_key))
+        if not request.json()['success']:
+            stderr(str(request.json()['error']))
+            return bot.reply('Sorry, something went wrong')
+    else:
+        request = requests.get(fiat_url)
+
+    request.raise_for_status()
+    rates_fiat_json = request.json()
+    rates_fiat_json['date'] = time.time()
+    rates_fiat_json['rates']['EUR'] = 1.0  # Put this here to make logic easier
+
+
+def get_rate(of, to):
+    global rates_fiat_json
+    of = of.upper()
+    to = to.upper()
+
+    if of == 'BTC' or to == 'BTC':
+        if of == 'BTC':
+            code = to
+            reverse = False
+        else:
+            code = of
+            reverse = True
+
+        return btc_rate(code, reverse)
+
+    if of not in rates_fiat_json['rates']:
+        return "Sorry {} isn't currently supported".format(of)
+
+    if to not in rates_fiat_json['rates']:
+        return "Sorry {} isn't currently supported".format(to)
+
+    return (1 / rates_fiat_json['rates'][of]) * rates_fiat_json['rates'][to]
+
+
+def exchange(bot, match):
     """Show the exchange rate between two currencies"""
-    if not trigger.group(2):
-        return bot.reply("No search term. An example: .cur 20 EUR in USD")
-    match = regex.match(trigger.group(2))
+
+    # match = regex.match(trigger)                                                      # that
     if not match:
-        # It's apologetic, because it's using Canadian data.
         bot.reply("Sorry, I didn't understand the input.")
         return NOLIMIT
 
-    amount, of, to = match.groups()
+    update_rates(bot)  # Try and update rates. Rate-limiting is done in update_rates()
+
+    query = match.string
+
+    others = query.split()
+    amount = others.pop(0)
+    of = others.pop(0)
+    others.pop(0)
+
+#    amount, of, _, *others = query.split() # I'd much rather use this, but it's not python 2.7 compatible
+
     try:
         amount = float(amount)
     except ValueError:
         bot.reply("Sorry, I didn't understand the input.")
     except OverflowError:
         bot.reply("Sorry, input amount was out of range.")
-    display(bot, amount, of, to)
+
+    out_string = '{} {} is'.format(amount, of.upper())
+    for to in others:
+        try:
+            out_string = build_reply(bot, amount, of.upper(), to.upper(), out_string)
+        except (requests.exceptions.HTTPError, requests.exceptions.ConnectionError) as e:
+            bot.reply("Something went wrong while I was getting the exchange rate.")
+            stderr("Error in GET request: {}".format(e))
```

Comment:
Another `stderr` call that should be converted to use the logger.

Reply:
Moved to a LOGGER.error()